src/go/k8s/pkg/resources/certmanager/type_helpers.go

```diff
@@ -251,6 +252,9 @@ func (cc *ClusterCertificates) prepareAPI(
 		return tlsDisabledAPICertificates(), nil
 	}
 	result := tlsEnabledAPICertificates(cc.pandaCluster.Namespace)
+	if internalTLSListener != nil {
+		result.internalTlsEnabled = true
```

Comment:
I don't know the code very well, but perhaps:
```suggestion
		 if internalTLSListener != nil {
			tls = internalTLSListener.GetTLS()
			if tls !=- nil {
				result.internalTlsEnabled = tls.Enabled
			}
		}
```

Reply:
```
func getInternalTLSListener(listeners []APIListener) APIListener {
	tlsListeners := getTLSListeners(listeners)
	for _, l := range tlsListeners {
		if ext := l.GetExternal(); ext == nil || !ext.Enabled {
			return l
		}
	}
	return nil
}

func getTLSListeners(listeners []APIListener) []APIListener {
	res := []APIListener{}
	for i, el := range listeners {
		tlsConfig := el.GetTLS()
		if tlsConfig != nil && tlsConfig.Enabled {
			res = append(res, listeners[i])
		}
	}
	return res
}
```

it's already handled here